indico/web/client/js/react/components/WTFPrivacyPolicyURLsField.jsx

```diff
@@ -0,0 +1,136 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useMemo, useState, useEffect} from 'react';
+
+import {Translate} from 'indico/react/i18n';
+
+export default function WTFPrivacyPolicyURLsField({fieldId, wrapperId, initialValues}) {
+  const parentElement = useMemo(() => document.getElementById(wrapperId), [wrapperId]);
+  const [values, setValues] = useState(
+    initialValues.length === 0
+      ? [{id: 0, title: '', url: ''}]
+      : initialValues.map((value, index) => ({
+          id: index,
+          title: value.title,
+          url: value.url,
+        }))
+  );
+
+  // Trigger change only after the DOM has changed
+  useEffect(() => {
+    parentElement.dispatchEvent(new Event('change', {bubbles: true}));
+  }, [values, parentElement]);
+
+  const handleChange = (id, field) => () => {
+    const idx = values.findIndex(value => value.id === id);
+    const newValues = [...values];
+    newValues[idx][field] = document.getElementById(`${id}-${field}`).value;
+    setValues(newValues);
+  };
+
+  const getProcessedValues = () => {
+    const newValues = values.filter(value => value.title !== '' || value.url !== '');
+    if (newValues.length === 1) {
+      return [{title: '', url: newValues[0].url}];
+    }
+    return newValues.map(value => ({
```

Comment:
This looks like it's just copying newValues...

Reply:
That is for removing the `id` attribute, which is only used for the front-end.. Might not be necessary since the back-end won't store it anyway

Reply:
No, we shouldn't send unnecessary data to the backend. But I think with my other suggested changes we may not need this ID at all?